### DISCUSSION TITLE
em7455 dell brand flashing didn't work

### DISCUSSION THREAD

**Discussion by ghost** [danielewood/sierra-wireless-modems#60] [Q&A]
Hi. I don't even know how to start it because everything seemed fine amd there weren't any errors. I was using your autoflash script on the newest Ubuntu (because not so long ago you updated it). I'll try tomorrow to flash it using old ubuntu, but I thought that it woule be good to write this. Well, how to start...
First I booted into windows to check if the adapter even works and indeed it did. However as it isn't the laptop that was meant for this card (dw5811e) it didn't let me install the drivers so it was just visible in device manager. When powered there was always blue led visible from the adapter (i think that's the identical model to yours). So next I downloaded Ubuntu 20.10 and booted to it after uninstalling the card from device manager (not that it was usable, but I read somewhere that this card can sometimes revert when they see the old settings).

On the ubuntu I connected the modem and pasted your code. And it started going and after maybe not even 10 minutes it was done, there was blue light flashing on and off and after the script was done it stopped glowing completely. So I turned the pc off, disconnected modem and booted to windows. Unfortunately when I connected it it was still visible in device manager under the old name and the power led only briefly flashed for a second. So I checked your site again and found out in the comments that you are supposed to run this as an admin. Didn't believe that this was the case because I saw the video and it was running normally, but booted again anyway. On the ubuntu the light was still off, but it returned few times during the script, even flashing on and off again. Unfortunately it still didn't work. I tried to connect to internet using mobile broadband that was available on ubuntu, but couldn't even add the settings so I restarted it and this time I could anter settings, but it doesn't let me connect. Also the card is still listed as dell. 
Do you have any ideas what I need to do now? Or any hints where to search and for what? Are there any logs that might tell you what went wrong? I mean, not the logs from these two installations, these are gone since it was running from love usb, but I can make another and I even want to try the old 18.04 ubuntu, but I don't believe it will work.

Have a good day and hopefully you can respond soon

**Comment by danielewood** [comment]
Pretty much impossible to guess given that there are no logs. 

Sudo runs the script as admin and it has safeguards to prevent running as anything other than root(or sudo root). 

Without logs, there really is nothing to do.

**Reply by ghost** [reply]
That's a shame. Can I at least somehow save the terminal output to a txt file so I can post it here? Maybe I missed something. I mean, the script must have done at least one thing because of the power light.

**Reply by ghost** [reply]
Update:
I flashed it using your old version of autoflash from 3 months ago and it works. I used old Ubuntu 18.04.5 and changed the link in wget from master to patch-1 and all worked. I then booted the newest Ubuntu 20.10 and connected to the network using the modem. And there were no problems. So I have no idea what was wrong, but it works so we can mark it as solved.

**Comment by sierra66** [comment]
I had troubles to flash mine some time ago, my first tries had removed all FW from the board, then flashing new firmware finished with an ok (seen the led flashing, and the upload progress on the terminal), but in the end, no firmware were stored on the device. What finally worked for me was to use `fwdwl-litehostx86_64` **without** the `--dmreset` option ???

```
#!/bin/bash


devpath=`ls /dev | grep -i -E 'cdc-wdm|qcqmi'`
devtype=`expr "$devpath" : '\(cdc-wdm\|qcqmi\)[0-9]$'`
case $devtype in
    cdc-wdm) devtype="MBIM" ;;
    qcqmi) devtype="QMI" ;;
    *) printf "Unknown Device Type = $devtype\r\nDevice Path = /dev/$devpath\r\n"; exit
esac
printf "Device Type = $devtype\r\nDevice Path = /dev/$devpath\r\n"

./fwdwl-litehostx86_64 \
--devmode $devtype  \
--devpath /dev/$devpath \
--modelfamily 3 \
--logfile "fwdwl-lite-$devpath.log" \
-e 1 \
--fwpath "./fw/" \
#--dmreset
```
